### ISSUE TITLE
Add MaxWait on jetstreamcontext.Subscribe(and similar) 

### ISSUE THREAD

**Issue by tpihl** [nats-io/nats.go#699]
## Feature Request
Creating an ephemeral or durable push subscriber using jetstreamcontext allow you to set almost all consumer parameters, but i cannot find MaxWait(dur).

**Comment by wallyqs** [comment]
@tpihl Right now there is no `nats.MaxWait(time.Duration)` for Subscribe which we should add, though just to be clear you can set `js.SubscribeSync("foo", nats.AckWait(1o*time.Duration))` to set how long to wait before the redelivery of a message:
https://github.com/nats-io/nats.go/blob/master/js.go#L771

**Comment by tpihl** [comment]
Thats not really what i meant. 

The MaxWait i am looking for is the push-consumers timeout before it will send an un-acked message. Defaults to 30 seconds, visible when you do a nats consumer info <stream> <consumer> as Ack Wait (under Ack Policy). How do i change that when i create a push consumer subscription as above?

**Comment by wallyqs** [comment]
If you call this:
```
js.SubscribeSync("foo", nats.AckWait(10*time.Duration))
```

Then the push consumer would redeliver messages after they have not been acked after 10 seconds.

**Comment by tpihl** [comment]
So MaxWait consumer require a SubscribeSync with my own msg-loop? Isnt that a pull-consumer?

**Comment by wallyqs** [comment]
`SubscribeSync` can be used to process synchronously although it is a push based consumer so the messages will all be delivered and be ready in the buffer from the client as they are delivered (max 8192 msgs buffered by default / 65536 for async subscribers).  With PullSubscriber, the messages are only delivered when `sub.Fetch(int)` is called.

**Comment by tpihl** [comment]
So what is the difference except max messages in buffer. My MsgHandler is only invoked one message at the time,isn't it?

**Comment by wallyqs** [comment]
Yes the MsgHandler is only invoked once per message delivered and it is called sequentially as the messages get delivered to the client, the main difference with PullSubscribers is that you control when to be delivered the messages whereas with Push conssumer, SubscribeSync or Subscribe it is expected that a client is always able to receive messages at a good pace without becoming a slow consumer.
